File: src/main/kotlin/io/github/dqualizer/dqlang/types/dam/architecture/apischema/APISchema.java
package io.github.dqualizer.dqlang.types.dam.architecture.apischema;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.dam.Identifiable;
import io.github.dqualizer.dqlang.types.dam.architecture.apischema.data.DataSchema;
import io.github.dqualizer.dqlang.types.dam.architecture.apischema.field.FieldItem;
import lombok.*;

import java.lang.module.ModuleDescriptor;
import java.util.Map;
import java.util.Set;

/**
 * General API-schema description
 */
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class APISchema extends Identifiable {

    private ModuleDescriptor.Version version;
    private String context;
    private String api;
    @JsonProperty("server_info")
    private Set<ServerInfo> serverInfo;
    private Set<FieldItem> field;
    @JsonProperty("data_schemas")
    private Map<String, DataSchema> dataSchemas;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/dam/architecture/apischema/APISchema.kt
package io.github.dqualizer.dqlang.types.dam.architecture.apischema

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.dam.Identifiable
import io.github.dqualizer.dqlang.types.dam.architecture.apischema.data.DataSchema
import io.github.dqualizer.dqlang.types.dam.architecture.apischema.field.FieldItem
import lombok.*
import java.lang.module.ModuleDescriptor

data class APISchema(
    val version: ModuleDescriptor.Version? = null,

    val context: String? = null,

    val api: String? = null,

    @JsonProperty("server_info")
    val serverInfo: Set<ServerInfo>? = HashSet(),

    val field: Set<FieldItem>? = HashSet(),

    @JsonProperty("data_schemas")
    private val dataSchemas: Map<String, DataSchema>? = HashMap()
): Identifiable()

